com.unity.netcode.gameobjects/Runtime/Core/SnapshotSystem.cs

```diff
@@ -937,6 +960,23 @@ internal void ReadSnapshot(ulong clientId, Stream snapshotStream)
                 var sequence = reader.ReadUInt16();
 
                 // todo: check we didn't miss any and deal with gaps
+
+                if (m_ClientData[clientId].ReceivedSequenceMask != 0)
+                {
+                    // since each bit in ReceivedSequenceMask is relative to the last received sequence
+                    // we need to shift all the bits by the difference in sequence
+                    m_ClientData[clientId].ReceivedSequenceMask <<=
+                        (sequence - m_ClientData[clientId].LastReceivedSequence);
```

Comment:
Do we need to to bounds checking on `sequence - m_ClientData[clientId].LastReceivedSequence` to guard against it being negative or overflowing the 16 bit value?